app/http/httpclient.c

```diff
@@ -292,33 +284,50 @@ static void ICACHE_FLASH_ATTR http_connect_callback( void * arg )
 
 	if ( req->post_data != NULL ) /* If there is data then add Content-Length header. */
 	{
-		os_sprintf( post_headers, "Content-Length: %d\r\n", strlen( req->post_data ) );
+		os_sprintf( post_headers, "Content-Length:%d\r\n", strlen( req->post_data ) );
 	}
 
-	if(req->headers == NULL) /* Avoid NULL pointer, it may cause exception */
+	if (req->headers == NULL) /* Avoid NULL pointer, it may cause exception */
 	{
 		req->headers = (char *)os_malloc(sizeof(char));
 		req->headers[0] = '\0';
 	}
 
+	char ua_header[32] = "";
+	int ua_len = 0;
+	
+	if ( os_strstr( req->headers, "User-Agent:" ) == NULL )
```

Comment:
I didn't see stricmp in any of the header files that I looked in. In this case, since it's data provided by the Lua developer (and not some arbitrary internet server), I think it's okay to just check the two cases: `User-Agent` and `user-agent`.